File: src/main/java/com/jozufozu/exnihiloomnia/common/registries/command/CommandRegistry.java
package com.jozufozu.exnihiloomnia.common.registries.command;

import com.jozufozu.exnihiloomnia.common.registries.ReloadableRegistry;
import net.minecraft.command.CommandBase;
import net.minecraft.command.CommandException;
import net.minecraft.command.ICommandSender;
import net.minecraft.command.SyntaxErrorException;
import net.minecraft.server.MinecraftServer;
import net.minecraft.util.math.BlockPos;

import javax.annotation.Nullable;
import java.util.ArrayList;
import java.util.List;

public class CommandRegistry extends CommandBase
{
    @Override
    public String getName()
    {
        return "enoreg";
    }
    
    @Override
    public String getUsage(ICommandSender sender)
    {
        return "commands.exnihiloomnia.enoreg.usage";
    }
    
    @Override
    public void execute(MinecraftServer server, ICommandSender sender, String[] args) throws CommandException
    {
        if (args.length != 2)
        {
            throw new SyntaxErrorException("commands.exnihiloomnia.enoreg.usage");
        }
        
        boolean all = "all".equals(args[1]);
        
        long start = System.nanoTime();
        
        int count = 0;
        
        for (ReloadableRegistry registry : ReloadableRegistry.getRegistries())
        {
            String resourcePath = registry.registryName.getResourcePath();
            if (all || resourcePath.equals(args[1]))
            {
                if ("reload".equals(args[0]))
                {
                    registry.load();
                }
                else if ("clear".equals(args[0]))
                {
                    registry.clear();
                }
                count++;
            }
        }
    
        double seconds = ((double) System.nanoTime() - start) * 1E-9;
        
        notifyCommandListener(sender, this, "commands.exnihiloomnia.enoreg.done", count, seconds);
    }
    
    @Override
    public List<String> getTabCompletions(MinecraftServer server, ICommandSender sender, String[] args, @Nullable BlockPos targetPos)
    {
        if (args.length == 1)
        {
            return getListOfStringsMatchingLastWord(args,"reload", "clear");
        }
        if (args.length == 2)
        {
            List<String> completions = new ArrayList<>();
            completions.add("all");
            
            for (ReloadableRegistry registry : ReloadableRegistry.getRegistries())
            {
                completions.add(registry.registryName.getResourcePath());
            }
            
            return getListOfStringsMatchingLastWord(args, completions);
        }
        
        return super.getTabCompletions(server, sender, args, targetPos);
    }
}

File: src/main/kotlin/com/jozufozu/exnihiloomnia/common/commands/CommandRegistry.kt
package com.jozufozu.exnihiloomnia.common.commands

import com.jozufozu.exnihiloomnia.common.registries.RegistryManager
import com.jozufozu.exnihiloomnia.common.registries.ReloadableRegistry
import net.minecraft.command.CommandBase
import net.minecraft.command.CommandException
import net.minecraft.command.ICommandSender
import net.minecraft.command.SyntaxErrorException
import net.minecraft.server.MinecraftServer
import net.minecraft.util.math.BlockPos
import java.util.*

class CommandRegistry : CommandBase() {
    override fun getName(): String {
        return "enoregistry"
    }

    override fun getUsage(sender: ICommandSender): String {
        return "commands.exnihiloomnia.enoreg.usage"
    }

    @Throws(CommandException::class)
    override fun execute(server: MinecraftServer, sender: ICommandSender, args: Array<String>) {
        if (args.size != 2) {
            throw SyntaxErrorException("commands.exnihiloomnia.enoreg.usage")
        }

        val all = "all" == args[1]

        val start = System.nanoTime()

        var count = 0

        for (registry in ReloadableRegistry.getRegistries()) {
            if (registry == RegistryManager.ORES) continue

            val resourcePath = registry.registryName.resourcePath
            if (all || resourcePath in args.slice(1 until args.size)) {
                if ("reload" == args[0]) {
                    registry.load()
                } else if ("clear" == args[0]) {
                    registry.clear()
                }
                count++
            }
        }

        val seconds = (System.nanoTime().toDouble() - start) * 1E-9

        CommandBase.notifyCommandListener(sender, this, "commands.exnihiloomnia.enoreg.done", count, seconds)
    }

    override fun getTabCompletions(server: MinecraftServer, sender: ICommandSender, args: Array<String>, targetPos: BlockPos?): List<String> {
        if (args.size == 1) {
            return CommandBase.getListOfStringsMatchingLastWord(args, "reload", "clear")
        }
        if (args.size == 2) {
            val completions = ArrayList<String>()
            completions.add("all")

            for (registry in ReloadableRegistry.getRegistries()) {
                completions.add(registry.registryName.resourcePath)
            }

            return CommandBase.getListOfStringsMatchingLastWord(args, completions)
        }

        return super.getTabCompletions(server, sender, args, targetPos)
    }
}
